src/Compilers/CSharp/Test/Emit/CodeGen/SwitchTests.cs

```diff
@@ -9718,5 +9718,1207 @@ .locals init (int V_0)
         }
 
         #endregion "regression tests"
+
+        #region Code Quality tests
+
+        [Fact]
+        public void BalancedSwitchDispatch_Double()
+        {
+            var source = @"using System;
+class C
+{
+    static void Main()
+    {
+        Console.WriteLine(M(2.1D));
+        Console.WriteLine(M(3.1D));
+        Console.WriteLine(M(4.1D));
+        Console.WriteLine(M(5.1D));
+        Console.WriteLine(M(6.1D));
+        Console.WriteLine(M(7.1D));
+        Console.WriteLine(M(8.1D));
+        Console.WriteLine(M(9.1D));
+        Console.WriteLine(M(10.1D));
+        Console.WriteLine(M(11.1D));
+        Console.WriteLine(M(12.1D));
+        Console.WriteLine(M(13.1D));
+        Console.WriteLine(M(14.1D));
+        Console.WriteLine(M(15.1D));
+        Console.WriteLine(M(16.1D));
+        Console.WriteLine(M(17.1D));
+        Console.WriteLine(M(18.1D));
+        Console.WriteLine(M(19.1D));
+        Console.WriteLine(M(20.1D));
+        Console.WriteLine(M(21.1D));
+        Console.WriteLine(M(22.1D));
+        Console.WriteLine(M(23.1D));
+        Console.WriteLine(M(24.1D));
+        Console.WriteLine(M(25.1D));
+        Console.WriteLine(M(26.1D));
+        Console.WriteLine(M(27.1D));
+        Console.WriteLine(M(28.1D));
+        Console.WriteLine(M(29.1D));
+
+    }
+    static int M(double d)
+    {
+        return d switch
+        {
+            >= 27.1D and < 29.1D => 19,
```

Comment:
nit: I am pretty confused by the numbering scheme, which seems random to my eyes. Why not just say `>= 27.1D and < 29.1D => 27`  (using one of the bounds as the identifier for the interval)?

Reply:
This test was automatically generated by the fuzz test  `PatternMatchingTests3.RelationalFuzz_01`.  The result numbering is in the order of the numbers the cases can match.

---
In reply to: [400466779](https://github.com/dotnet/roslyn/pull/42313#discussion_r400466779) [](ancestors = 400466779)